installers/cubeit-installer

```diff
@@ -1108,6 +1108,13 @@ if [ -n "${HDINSTALL_CONTAINERS}" ]; then
 	    ${f}
 	done
     fi
+
+    #prepare the firmwares in essential
+    if [ ! -d ${TMPMNT}/var/lib/cube/essential/lib/firmware ]; then
+        mkdir -p ${TMPMNT}/var/lib/cube/essential/lib/firmware
+    fi
+    cp -a ${TMPMNT}/opt/container/dom0/rootfs/lib/firmware/* ${TMPMNT}/var/lib/cube/essential/lib/firmware
```

Comment:
This needs to be controlled by a configuration flag, and not be a always on feature.  We do not want this to be the default mode of operation.

We really need to understand the target, what firmware it needs, and only copy that firmware.

Reply:
Hi, Bruce

The  script of https://github.com/OverC/meta-overc/blob/master-oci/meta-cube/recipes-support/dom0-init/files/firmware-sync didn't do the target specific and just sync between dom0's firmware with essential's dir /var/lib/cube/essential/lib/firmware, why we should care about the target in the installer? This codes was only do those sync ahead.

Reply:
That sync is going to have intelligence added to it shortly, along with statistics gathering:

   - tracking of used firmware
   - conditional and partial sync of firmware options
   - 3rd party firmware sync from alternate containers

The point is that we can't have the installer always putting the firmware in place, any more than we can have it always baked into essential. Both of those options break the ability to manage the firmware in a flexible manner.